File: Project/Project/microservices/result-service/src/main/java/com/certifypro/result/service/ResultService.java
package com.certifypro.result.service;

import com.certifypro.result.dto.response.CandidateResultResponse;
import com.certifypro.result.dto.response.PageResponse;

import java.util.List;

public interface ResultService {

    List<CandidateResultResponse> compute(Long windowId);

    CandidateResultResponse publish(Long resultId);

    PageResponse<CandidateResultResponse> view(Long candidateId, Long windowId, int page, int limit);
}

File: Project/Project/frontend/src/services/resultService.js
import http from './http';

export async function compute(windowId) {
  const res = await http.post(`/results/compute/${windowId}`);
  return res.data.data;
}
export async function publish(id) {
  const res = await http.patch(`/results/${id}/publish`);
  return res.data.data;
}
export async function view(params) {
  const res = await http.get('/results', { params });
  return res.data.data;
}
